docs/docs/building-ui/composite-components.md

```diff
@@ -1,58 +1,146 @@
 ---
 sidebar_position: 2
 title: Composite Components
-draft: false
+sidebar_class_name: updated-content
 ---
 
 <DocChip chip='since' label='23.06' />
 <JavadocLink type="foundation" location="com/webforj/component/Composite" top='true'/>
 
-Developers will often wish to create components that contain constituent components for application level use. The `Composite` component gives developers the tools they need to create their own components while maintaining control over what they choose to expose to users. 
+The `Composite` component in webforJ lets you create custom, self-contained components by wrapping internal components into a single reusable unit. You can encapsulate logic, layout, and styling while controlling exactly what methods and data is exposed.
 
-It allows developers to manage a specific type of `Component` instance, providing a way to encapsulate its behavior. It requires any extending subclass to specify the type of `Component` it intends to manage, ensuring a subclass of `Composite` is intrinsically linked to its underlying `Component`.
+Use a `Composite` when you want to:
+	-	Reuse component patterns throughout your app
+	-	Encapsulate layout and behavior into a single, maintainable unit
+	-	Compose multiple components without exposing implementation details
 
-:::tip
 It's highly recommended to create custom components by utilizing the `Composite` component, rather than extending the base `Component` component.
-:::
 
-To utilize the `Composite` component, start by creating a new Java class that extends the `Composite` component. Specify the type of Component you want to manage as the generic type parameter.
+## Defining a `Composite`
+
+To define a `Composite` component, extend the `Composite` class and specify the type of component it manages. This becomes your bound component, which is the root container that holds your internal structure. The bound component can be any [webforJ component](../components/overview) or [HTML element component](/docs/building-ui/web-components/html-elements).
 
 ```java
 public class ApplicationComponent extends Composite<Div> {
 	//Implementation
 }
 ```
 
-## Component binding
+## Constructor setup (Recommended)
+
+The recommended approach is to configure everything in the constructor. Use `getBoundComponent()` to build your internal layout and logic directly.
```

Comment:
Where is this recommendation coming from? It should say that you can configure the component either in the constructor or in the onDidCreate method. see https://docs.webforj.com/docs/building-ui/composite-components#lifecycle-management